### PR TITLE
Redux in react

### PR DESCRIPTION
**Pull request by richardoppiyo**
In this ,milestone:
- I implemented Provider and  store in the main component of the  React app.
- I added funcionality for "Add new" button click
- added a funcionality for "Remove" button clic
-added a funcionality for "Check status" button click:

### CODE REVIEW

File: src/components/InputBook.js [hunk 1]
```diff
@@ -1,25 +1,46 @@
...
-        placeholder="Author"
-        name="title"
-      />
-      <button type="submit" className="input-submit">Submit</button>
-    </form>
-  </>
-);
+const InputBook = () => {
+  const dispatch = useDispatch();
+  const [title, setTitle] = useState('');
```
Comment (micheaol) [inline src/components/InputBook.js]: Please ensure that your ` InputBook.js` has a local state with  `title` and `author` according to the project requirement.

File: src/components/InputBook.js [hunk 2]
```diff
@@ -1,25 +1,46 @@
...
+
+  const submitBookToStore = (e) => {
+    e.preventDefault();
+    const newBook = {
+      id: uuidv4(),
+      title,
+    };
+    console.log(newBook);
```
Comment (micheaol) [inline src/components/InputBook.js]: Please remove `console.log` from your codebase.

File: src/components/Book.js [hunk 3]
```diff
@@ -1,25 +1,62 @@
...
+//     </div>
+//   </div>
+// ))}
+// const Book = () => {
+//   const books = useSelector((state) => state.book);
+//   return (
+//     <div className="book">
+//       <div className="book-holder">
+//         <h2>
+//           Title:
```
Comment (micheaol) [inline src/components/Book.js]: Please remove commented code from your codebase, to make your codebase clean.

File: src/components/Categories.js [hunk 4]
```diff
@@ -1,12 +1,50 @@
...
+
+// const Categories = () => {
+//   const status = useSelector((state) => state.categReducers);
+//   const dispatch = useDispatch();
+//     <>
+//       <Header />
+//       <button className="book-button" type="button" onClick={() =>
+//  { dispatch(checkStatus()); }}>Check Status</button>
+//       <p>{status}</p>
+//     </>;
```
Comment (micheaol) [inline src/components/Categories.js]: Please remove commented code from your codebase, to make your codebase clean.

File: src/components/Categories.js [hunk 5]
```diff
@@ -1,12 +1,50 @@
...
+// };
+
+function Categories() {
+  const status = useSelector((state) => state.categ);
+  const dispatch = useDispatch();
+  return (
+    <div>
+      <Header />
+      <button className="book-button" type="button" onClick={() => { dispatch(checkStatus()); }}>Check Status</button>
```
Comment (micheaol) [inline src/components/Categories.js:13]: Please ensure that the `checkCategoriesStatus` dispatches the `underConstruction` action creator to the screen. At the moment, nothing shows on the screen. Kindly fix that.

**Review by micheaol — CHANGES_REQUESTED** [review]
Hi @richardoppiyo , 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by Kalunge — APPROVED** [review]
Hi, @richardoppiyo >, 

You have done well to implement all the changes requested by the previous reviewer. 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

[OPTIONAL]
- There is an error in the console for the  id provided from UUID as it is not unique, please check and resolve that error


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**